core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -394,6 +445,10 @@ private LoadTableResponse loadInternal(
 
   @Override
   public Table loadTable(SessionContext context, TableIdentifier identifier) {
+    if (!endpoints.contains(Endpoint.V1_LOAD_TABLE)) {
+      throw new NoSuchTableException("Unable to load table %s.%s", name(), identifier);
```

Comment:
I don't feel like this message is right.  Shouldn't we say "Loading tables is not supported by the Catalog"?  I'm not sure if this really matters too much since pretty much everything requires table loading.

Reply:
It also looks like we have a few places where we're using `check` vs `contains` because we need to throw a specific exception.  What if we just make this use a supplier so that we can be consistent about checks:

```java
Endpoint.check(endpoints, Endpoint.V1_LOAD_TABLE, () -> new NoSuchTableException(...));
```

Reply:
> I don't feel like this message is right. Shouldn't we say "Loading tables is not supported by the Catalog"? I'm not sure if this really matters too much since pretty much everything requires table loading.

I had this in my initial implementation but changed it after https://github.com/apache/iceberg/pull/10929#discussion_r1715632323.

Reply:
I've applied both of your suggestions now